added light brightness support

Sjack-Sch opened the pull request:
Addition to #57.
Home Assistant brightness (0-255) will be converted to Home Connect brightness (10-100) and vice versa.

GrumpyMeow commented:
Hi David,
could you maybe merge this Pull Request?

DavidMStraub commented on custom_components/home_connect_beta/light.py:
```diff
@@ -1,9 +1,11 @@
 from homeconnect.api import HomeConnectError
 
-from homeassistant.components.light import LightEntity
+from homeassistant.components.light import (ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, LightEntity)
```
Can you please apply `isort` and then format with `black`? (This is required for HA Core)

Sjack-Sch replied:
OK

DavidMStraub commented on custom_components/home_connect_beta/light.py:
```diff
@@ -38,22 +40,48 @@ def __init__(self, device, desc):
+        """Switch the light on / change brightness."""
+        if ATTR_BRIGHTNESS in kwargs:
+            _LOGGER.debug("Tried to change brightness for: %s", self.name)
+            """ Convert Home Assistant brightness (0-255) to Home Connect brightness (10-100)
```
Can you please modify this doc string to make `pydocstyle` happy? (This is required for HA core)

Sjack-Sch replied:
HI David, not sure what to modify here. I used the tool pydocstyle, but it did not detect any unconformities.

Sjack-Sch replied:
Done.

DavidMStraub commented on custom_components/home_connect_beta/light.py:
```diff
@@ -82,4 +110,10 @@ async def async_update(self):
-        _LOGGER.debug("Updated, new light state: %s", self._state)
\ No newline at end of file
+        brightness = self.device.appliance.status.get(COOKING_LIGHTINGBRIGHTNESS, {})
+        if not brightness:
```
I think this should be `if brightness is None` because 0 is a valid value and shouldn't lead to the state being shown as "unavailable".

Sjack-Sch replied:
Home connect returns a value from 10 up to and including 100 with steps of 1.
0 is never returned. Better to change to `if brightness is None` for consistency reasons?

DavidMStraub replied:
Fair enough, but yes, I would suggest to do it for consistency.

Sjack-Sch replied:
Updated and committed.

DavidMStraub commented:
Looks great! Just a couple of minor comments above.

DavidMStraub commented:
> could you maybe merge this Pull Request?

Yeah sorry I somehow missed it.

GrumpyMeow commented:
https://github.com/Sjack-Sch/homeassistant-homeconnect/pull/1

Sjack-Sch commented:
> I implemented AmbientLight control:
> [Sjack-Sch#1](https://github.com/Sjack-Sch/homeassistant-homeconnect/pull/1)

Thanks, shall we wait merging ambient until brightness is merged?
In the mean time I can test your version on a hood without ambient support.

DavidMStraub commented:
Done?

Sjack-Sch commented:
> Done?

Yep

Sjack-Sch commented:
> I implemented AmbientLight control:
> [Sjack-Sch#1](https://github.com/Sjack-Sch/homeassistant-homeconnect/pull/1)

GrumpyMeow, could you please create a pull request from DavidMStraub/homeassistant-homeconnect to add your AmbientLight control?

anthonyangel commented:
> > I implemented AmbientLight control:
> > [Sjack-Sch#1](https://github.com/Sjack-Sch/homeassistant-homeconnect/pull/1)
> 
> GrumpyMeow, could you please create a pull request from DavidMStraub/homeassistant-homeconnect to add your AmbientLight control?

I loaded up the code in that PR to test, the ambient light (including on/off, colour & brightness) works as expected for a hood, but not for a dishwasher, I was getting the error: `Error while trying to turn on light: {'key': 'SDK.Error.UnsupportedSetting', 'description': 'Setting is not supported'}` (I can't think of when I'd use the dishwasher light, but as it was there I figured that I'd give it a try anyway.

There was another issue that the startup time of the home_connect_beta component was about 6 minutes, which is long

DavidMStraub commented:
Do you have an idea what it was doing in the 6 minutes?

anthonyangel commented:
I have 'Waiting on integrations to complete setup: home_connect_beta' entries in the logs, but not much else. I'm increasing the logging of 'homeassistant.bootstrap' to debug and will try again

Sjack-Sch commented:
> I implemented AmbientLight control:
> https://github.com/Sjack-Sch/homeassistant-homeconnect/pull/1

I added the ambient with #105.
Can you please help testing.

Sjack-Sch commented:
> > > I implemented AmbientLight control:
> > > [Sjack-Sch#1](https://github.com/Sjack-Sch/homeassistant-homeconnect/pull/1)
> > 
> > 
> > GrumpyMeow, could you please create a pull request from DavidMStraub/homeassistant-homeconnect to add your AmbientLight control?
> 
> I loaded up the code in that PR to test, the ambient light (including on/off, colour & brightness) works as expected for a hood, but not for a dishwasher, I was getting the error: `Error while trying to turn on light: {'key': 'SDK.Error.UnsupportedSetting', 'description': 'Setting is not supported'}` (I can't think of when I'd use the dishwasher light, but as it was there I figured that I'd give it a try anyway.
> 
> There was another issue that the startup time of the home_connect_beta component was about 6 minutes, which is long

Hi Anthony, Could you test #105, 
I added an additonal check: if the device has the Ambient feature before creating the Ambient light entity.

GrumpyMeow commented:
I’m on holiday for 2 weeks and this can’t help at this time with these changes. Sorry.
